### PR TITLE
Optionally expose the driver UI port as NodePort

### PR DESCRIPTION
**Pull request by kimoonkim**
## What changes were proposed in this pull request?

Addresses #129 by optionally exposing the driver Web UI as NodePort.

## How was this patch tested?

Tested again my k8s cluster. I was able to look at the Web UI using my browser.

Ran the integration test successfully:
```
[INFO] Spark Project Kubernetes Integration Tests ......... SUCCESS [15:47 min]
```

### CODE REVIEW

**Comment by mccheah** [comment]
Also modify `docs/running-on-kubernetes.md` - the table at the bottom of the page should include this configuration.

**Comment by kimoonkim** [comment]
Thanks @mccheah for taking a look. Updated `docs/running-on-kubernetes.md`.

**Comment by foxish** [comment]
Can we work out a better scheme for setting the type of the services we create? We have a similar need in https://github.com/apache-spark-on-k8s/spark/pull/116 as well.

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala
Comment (ash211) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala]: rename `serviceType` to `uiServiceType`

**Review by ash211 — COMMENTED** [review]
For a short-term addition this certainly gets the job done of exposing the driver UI port to outside the cluster.  For longer term I think we might want to have one flag for each service's type, in addition to a flag that controls all of them at once.

I'm not sure we should block this PR on that though.

@foxish do you have any strawman suggestions for a better scheme doing this?

**Comment by foxish** [comment]
>  I think we might want to have one flag for each service's type, in addition to a flag that controls all of them at once.

I was thinking along these same lines. This change makes sense for the short term.

**Comment by ash211** [comment]
Did the quick rename, good to merge when the build is green

**Review by ash211 — APPROVED** [review]

**Comment by kimoonkim** [comment]
Thanks for merging in this change.

@foxish, @ash211 It occurred to me that we may want to consider hierarchical config options for services that may be formatted in json or yaml, much like k8s resources. Otherwise, the flat key=value options may become messy pretty soon.

>I think we might want to have one flag for each service's type, in addition to a flag that controls all of them at once.

**Pull request merged**
